contracts/liquidity_hub/bonding-manager/src/commands.rs

```diff
@@ -0,0 +1,373 @@
+use cosmwasm_std::{
+    ensure, Addr, BankMsg, Coin, CosmosMsg, Decimal, DepsMut, Env, MessageInfo, Order, Response,
+    StdError, StdResult, Timestamp, Uint128, Uint64,
+};
+use white_whale_std::pool_network::asset;
+
+use white_whale_std::bonding_manager::Bond;
+
+use crate::helpers::validate_growth_rate;
+use crate::queries::{query_claimable, query_weight, MAX_PAGE_LIMIT};
+use crate::state::{
+    update_global_weight, update_local_weight, BOND, CONFIG, EPOCHS, GLOBAL, LAST_CLAIMED_EPOCH,
+    UNBOND,
+};
+use crate::{helpers, ContractError};
+
+/// Bonds the provided asset.
+pub(crate) fn bond(
+    mut deps: DepsMut,
+    timestamp: Timestamp,
+    info: MessageInfo,
+    _env: Env,
+    asset: Coin,
+) -> Result<Response, ContractError> {
+    helpers::validate_funds(&deps, &info, &asset, asset.denom.clone())?;
```

Comment:
This validate_funds function does nothing, what it's doing is to validate that the asset Coin matches what info is sending, but the former was derived from info, so it will always pass. The real validate funds function should be what is done before calling `bond`, i.e.

```
let config = CONFIG.load(deps.storage)?;
            // Ensure that the user has sent some funds
            ensure!(!info.funds.is_empty(), PaymentError::NoFunds {});
            let asset_to_bond = {
                // Filter the funds to include only those with accepted denominations
                let valid_funds: Vec<&Coin> = info
                    .funds
                    .iter()
                    .filter(|coin| config.bonding_assets.contains(&coin.denom))
                    .collect();

                // Check if there are no valid funds after filtering
                if valid_funds.is_empty() {
                    Err(PaymentError::MissingDenom("test".to_string()))
                } else if valid_funds.len() == 1 {
                    // If exactly one valid fund is found, return the amount
                    Ok(valid_funds[0])
                } else {
                    // If multiple valid denominations are found (which shouldn't happen), return an error
                    Err(PaymentError::MultipleDenoms {})
                }
            }?;
```